### ISSUE TITLE
What does the cycle stand for

### ISSUE THREAD

**Issue by liul-ing** [agaldran/lwnet#3]
Dear Author:
  I looked at the code several times and I don't understand what cycle means.I debug the code and found that there was only one image in training stage(function train_one_epoch) instead of batch_size:4.
  I know my question is easy,however i still can't understand after I debug the code,could you please anwer me?
Thanak you!!!

**Comment by agaldran** [comment]
Hey @liul-ing , I am sorry about the confusion. Let's see, when one trains with cyclical learning rate scheduling, what we do is to decide on a number of epochs that compose a single cycles, for example 10. Then, we start with an initial learning rate that slowly decays during the entire cycle, this is, during the 10 epochs, until it reaches a very low value, 0 or similar. Once this happens, the learning rate is reset to its initial value and a new cycle starts over.

I don't really understand what could be happening with the batch-size in your execution. I recenlty refactored this implementation to simplify it, and I might have messed things up. Can I ask what value do you see if you add in line 93 [here](https://github.com/agaldran/lwnet/blob/1c622a4dbbcd97a83087fee3e58403f9c1a516b9/train_cyclical.py#L93) a print of the shape of the inputs, `print(inputs.shape)`?

Thanks for using this code. Cheers,

Adrian
